packages/api/src/apps/datastore_api/app.ts

```diff
@@ -0,0 +1,327 @@
+        usd: {
+          latest: {
+            tvl: [0, "0"],
+          },
+          history: {
+            tvl: empStatsHistory.Table("Tvl Global History"),
+          },
+        },
+      },
+    },
+    lastBlockUpdate: 0,
+    registeredEmps: new Set<string>(),
+    registeredEmpsMetadata: new Map(),
+    registeredLsps: new Set<string>(),
+    registeredLspsMetadata: new Map(),
+    collateralAddresses: new Set<string>(),
+    syntheticAddresses: new Set<string>(),
+    // lsp related props. could be its own state object
+    longAddresses: new Set<string>(),
+    shortAddresses: new Set<string>(),
+    multicall2: new Multicall2(env.MULTI_CALL_2_ADDRESS, provider),
+    lsps: {
+      active: lsps.Table("Active LSP", datastores.lspsActiveStore),
+      expired: lsps.Table("Expired LSP", datastores.lspsExpiredStore),
+    },
+  };
+
+  // services for ingesting data
+  const services = {
+    // these services can optionally be configured with a config object, but currently they are undefined or have defaults
+    blocks: Services.Blocks(undefined, appState),
+    emps: Services.EmpState({ debug }, appState),
+    registry: await Services.Registry({ debug }, appState, (event, data) =>
+      serviceEvents.emit("empRegistry", event, data)
+    ),
+    collateralPrices: Services.CollateralPrices({ debug }, appState),
+    syntheticPrices: Services.SyntheticPrices(
+      {
+        debug,
+        cryptowatchApiKey: env.cryptowatchApiKey,
+        tradermadeApiKey: env.tradermadeApiKey,
+        quandlApiKey: env.quandlApiKey,
+        defipulseApiKey: env.defipulseApiKey,
+      },
+      appState
+    ),
+    erc20s: Services.Erc20s({ debug }, appState),
+    empStats: Services.stats.Emp({ debug }, appState),
+    marketPrices: Services.MarketPrices({ debug }, appState),
+    lspCreator: await Services.MultiLspCreator({ debug, addresses: lspCreatorAddresses }, appState, (event, data) =>
+      serviceEvents.emit("multiLspCreator", event, data)
+    ),
+    lsps: Services.LspState({ debug }, appState),
+    lspStats: Services.stats.Lsp({ debug }, appState),
+    globalStats: Services.stats.Global({ debug }, appState),
+  };
+
+  const initBlock = await provider.getBlock("latest");
+
+  // warm caches
+  await services.registry(appState.lastBlockUpdate, initBlock.number);
+  console.log("Got all EMP addresses");
+
+  await services.lspCreator.update(appState.lastBlockUpdate, initBlock.number);
+  console.log("Got all LSP addresses");
+
+  await services.emps.update(appState.lastBlockUpdate, initBlock.number);
+  console.log("Updated EMP state");
+
+  await services.lsps.update(appState.lastBlockUpdate, initBlock.number);
+  console.log("Updated LSP state");
+
+  // we've update our state based on latest block we queried
+  appState.lastBlockUpdate = initBlock.number;
+
+  await services.erc20s.update();
+  console.log("Updated tokens");
+
+  // backfill price histories, disable if not specified in env
+  if (env.backfillDays) {
```

Comment:
we are going to also have to deal with backfilling based on the latest data we know about vs the time the service starts. this also can be a separate issue

Reply:
I will make a separate PR for optimising the backfill and other function calls